wpe/src/main/java/com/wpe/wpe/Browser.java

```diff
@@ -49,6 +49,12 @@ public final class Browser {
      */
     private final BrowserGlue glue;
 
+    /**
+     * Under normal circumstances there is only one application context in a process, so it's safe
+     * to treat this as a global.
+     */
+    private static Context applicationContext;
```

Comment:
In Android, Context instances can drag quite a lot of resources and, depending on the type of context, can prevent the application to correctly stop.

Normally you should have a warning of type "Do not place Android context classes in static fields" in the Gradle build logs or the IDE.

At the end, we will still end-up by having a static instance somewhere as the Browser class is a singleton, but this is something we'll have to resolve differently using a custom Application class for example.

Reply:
not static anymore